sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts

```diff
@@ -332,8 +439,35 @@ export class ECertGenerationService {
         .reduce((previousValue, currentValue) => {
           return previousValue + currentValue.effectiveAmount;
         }, 0);
+      // Calculate total BC grants subtracted due to restriction.
+      let bcAmountSubtractedRestrictionId = NaN;
+      const bcTotalGrantAmountSubtracted =
+        disbursementSchedule.disbursementValues
+          // Filter all BC grants and filter out the null BC grants subtracted.
+          .filter((disbursementValue) => {
+            if (
+              disbursementValue.valueType === DisbursementValueType.BCGrant &&
+              disbursementValue.restrictionAmountSubtracted !== null
```

Comment:
Why do we need additional checks here? The idea of this method was to be concerned only about doing the SUM of the BC Grants if there is a value to be ignored it probably must be set to zero before this calculation happen.

Reply:
removed the logic got total bc grant